DEBUG-5433 DI: implicit enablement via remote configuration (#5525)
DataDog/dd-trace-rb: docs/GettingStarted.md

| Environment variable | Type | Description | Default |
|---|---|---|---|

> | `DD_DYNAMIC_INSTRUMENTATION_ENABLED` | `Boolean` | Controls Dynamic Instrumentation enablement. `true` enables at boot. `false` disables and blocks UI-driven enablement via remote configuration. Unset (the default) leaves DI off at boot but allows it to be turned on later from the Datadog UI when a probe is created. | `false` |

Reviewer: ```suggestion
| `DD_DYNAMIC_INSTRUMENTATION_ENABLED` | `Boolean` | Enables or disables Dynamic Instrumentation. `true` enables at boot; `false` disables and blocks UI-driven enablement through remote configuration. When unset (the default), Dynamic Instrumentation is off at boot but can be enabled from the Datadog UI when a probe is created. | `false` |
```

Author: Thanks for the review and the cleaner wording — pulled it in at 43c8b35698. "blocks UI-driven enablement through remote configuration" reads clearer than what was there before.
